pkg/asset/templates/kubelet.yaml

```diff
@@ -24,20 +24,60 @@ spec:
         - --
         - ./hyperkube
         - kubelet
-        - --api-servers={{ index .APIServers 0 }}
-        - --config=/etc/kubernetes/manifests
-        - --allow-privileged
+        - --api-servers=$(API_SERVERS)
+        - --config=$(CONFIG)
+        - --allow-privileged=$(ALLOW_PRIVILEGED)
         - --hostname-override=$(MY_POD_IP)
-        - --cluster-dns=10.3.0.10
-        - --cluster-domain=cluster.local
-        - --kubeconfig=/etc/kubernetes/kubeconfig
-        - --lock-file=/var/run/lock/kubelet.lock
-        - --minimum-container-ttl-duration=3m0s
+        - --cluster-dns=$(CLUSTER_DNS)
+        - --cluster-domain=$(CLUSTER_DOMAIN)
+        - --kubeconfig=$(KUBECONFIG)
+        - --lock-file=$(LOCK_FILE)
+        - --minimum-container-ttl-duration=$(MINIMUM_CONTAINER_TTL_DURATION)
         env:
           - name: MY_POD_IP
             valueFrom:
               fieldRef:
                 fieldPath: status.podIP
+          - name: API_SERVERS
```

Comment:
I would really like to not have to handle each flag as an individual entity.

We could just have the configmap drop a file which contains all flags (see example at: http://kubernetes.io/docs/user-guide/configmap)

Then we could just do something like: `bash -c "/kubelet $(cat $CONFIG)"`

We will eventually be replacing this with the upstream component config work ... so even if a file full of flags is a little gross, it is pretty clear how to use it.